compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/window/WindowStateTest.kt

```diff
@@ -601,28 +610,49 @@ class WindowStateTest {
         }
 
         awaitIdle()
-        assertThat(lastRecomposedState.placement).isEqualTo(initialState.placement)
-        assertThat(lastRecomposedState.isMinimized).isEqualTo(initialState.isMinimized)
-        assertThat(lastRecomposedState.size).isEqualTo(initialState.size)
-        assertThat(lastRecomposedState.position).isEqualTo(initialState.position)
+        assertWindowStateEquals(lastRecomposedState, initialState)
         lastRecomposedState.placement = newState.placement
         lastRecomposedState.isMinimized = newState.isMinimized
         lastRecomposedState.size = newState.size
         lastRecomposedState.position = newState.position
 
         index = 1
         awaitIdle()
-        assertThat(lastRecomposedState.placement).isEqualTo(initialState.placement)
-        assertThat(lastRecomposedState.isMinimized).isEqualTo(initialState.isMinimized)
-        assertThat(lastRecomposedState.size).isEqualTo(initialState.size)
-        assertThat(lastRecomposedState.position).isEqualTo(initialState.position)
+        assertWindowStateEquals(lastRecomposedState, initialState)
 
         index = 0
         awaitIdle()
-        assertThat(lastRecomposedState.placement).isEqualTo(newState.placement)
-        assertThat(lastRecomposedState.isMinimized).isEqualTo(newState.isMinimized)
-        assertThat(lastRecomposedState.size).isEqualTo(newState.size)
-        assertThat(lastRecomposedState.position).isEqualTo(newState.position)
+        assertWindowStateEquals(lastRecomposedState, newState)
+    }
+
+    @OptIn(ExperimentalTestApi::class)
+    @Test
+    fun `can save Unspecified window size`() = runComposeUiTest {
+        var initialState: WindowState? = null
+        var restoredState: WindowState? = null
+        var index by mutableIntStateOf(0)
+        setContent {
+            val saveableStateHolder = rememberSaveableStateHolder()
+            saveableStateHolder.SaveableStateProvider(index) {
+                val state = rememberWindowState(size = DpSize.Unspecified)
+                if (index == 0) {
+                    initialState = state
+                }
+                else {
+                    restoredState = state
+                }
+            }
+        }
+
+        index = 1
+        waitForIdle()
+        index = 2
```

Comment:
The test doesn't check that state was restored correctly.

To check that, we need to change the size, and switch back to `index = 0`

Reply:
I see the need to switch back to `index = 0`, but why change the size?

The latest commit seems to work (I purposely "broke" restoring in order to test it, and the test fails correctly).

Reply:
> but why change the size?

If we don't change the size, then we can't be sure that value was restored instead of reusing the initial passed value (or reused the value saved in index=1).

To check it correctly, seems we need to has initial size not Unspecified:
```
        var index by mutableIntStateOf(0)
        lateinit var lastRecomposedState: WindowState
        setContent {
            val saveableStateHolder = rememberSaveableStateHolder()
            saveableStateHolder.SaveableStateProvider(index) {
                val state = rememberWindowState(size = DpSize(10f, 10f))
                lastRecomposedState = sate
            }
        }

        state.size = DpSize.Unspecified
        waitForIdle()
        assertWindowStateEquals(WindowState(size = DpSize.Unspecified), lastRecomposedState)

        index = 1
        waitForIdle()
        assertWindowStateEquals(WindowState(size = DpSize(10f, 10f)), lastRecomposedState)

        index = 0
        waitForIdle()
        assertWindowStateEquals(WindowState(size = DpSize.Unspecified), lastRecomposedState)
```